queue/abstract/driver/utubettl.lua

```diff
@@ -87,18 +102,99 @@ function tube.create_space(space_name, opts)
         type = 'tree',
         parts = {i_status, str_type(), i_utube, str_type(), i_id, num_type()}
     })
+    space:create_index('utube_pri', {
+        type = 'tree',
+        parts = {i_status, str_type(), i_utube, str_type(), i_pri, num_type(), i_id, num_type()}
+    })
     return space
 end
 
 local delayed_state = { state.DELAYED }
 local ttl_states    = { state.READY, state.BURIED }
 local ttr_state     = { state.TAKEN }
 
+local function put_next_ready(self, utube)
+    local taken = self.space.index.utube:min{state.TAKEN, utube}
+    if taken == nil or taken[i_status] ~= state.TAKEN then
+        local next_task = self.space.index.utube_pri:min{state.READY, utube}
+        if next_task == nil or next_task[i_status] ~= state.READY then
+            return
+        end
+        -- Ignoring ER_TUPLE_FOUND error, if a tuple with the same task_id
+        -- or utube name is already in the space.
+        -- Note that both task_id and utube indexes are unique, so there will be
+        -- no duplicates: each task_id can occur in the space not more than once,
+        -- there can be no more than one task from each utube in a space.
+        pcall(self.space_ready_buffer.insert, self.space_ready_buffer, {next_task[i_id], utube, next_task[i_pri]})
+    end
+end
+
+local function put_ready(self, id, utube, pri)
+    local taken = self.space.index.utube:min{state.TAKEN, utube}
+    if taken == nil or taken[i_status] ~= state.TAKEN then
+        local current_task = self.space.index.utube_pri:min{state.READY, utube}
+        if current_task == nil or current_task[i_status] ~= state.READY or current_task[i_pri] < pri then
+            return
+        end
+        if current_task ~= nil and current_task[i_pri] > pri then
+            self.space_ready_buffer:delete(current_task[id])
+        end
+        -- Ignoring ER_TUPLE_FOUND error, if a tuple with the same task_id
+        -- or utube name is already in the space.
+        -- Note that both task_id and utube indexes are unique, so there will be
+        -- no duplicates: each task_id can occur in the space not more than once,
+        -- there can be no more than one task from each utube in a space.
+        pcall(self.space_ready_buffer.insert, self.space_ready_buffer, {id, utube, pri})
+    end
+end
+
+local function delete_ready(self, id, utube)
+    self.space_ready_buffer:delete(id)
+    put_next_ready(self, utube)
+end
+
+local function update_ready(self, utube, pri, id)
+    local prev_task = self.space_ready_buffer.index.utube:get{utube}
+    if prev_task ~= nil then
+        if prev_task[3] > pri or (prev_task[3] == pri and prev_task[1] > id) then
+            delete_ready(self, prev_task[1], utube)
```

Comment:
It seems too heavy for the update:
```suggestion
            self.space_ready_buffer:delete(id)
            self.space_ready_buffer:insert({id, utube, pri})
```